### ISSUE TITLE
Merge mining with Monero

### ISSUE THREAD

**Issue by SChernykh** [darkrenaissance/darkfi#244]
I've just found out about https://dark.fi/insights/development-update-q423.html and took a look at your merge mining proxy. It doesn't seem to be 100% compatbile with what me and @moneromooo-monero are trying to implement now.

Please read through https://github.com/SChernykh/p2pool/blob/merge-mining/docs/MERGE_MINING.MD (a WIP documentation) and https://github.com/monero-project/monero/pull/9073 (merge mining fixes for Monero).

As far as I can see, your mm proxy can't be used to mine multiple chains together right now, because it doesn't calculate the Merkle root hash or all merge mined chains (it's single chain only?).

**Comment by parazyd** [comment]
Hi. What the proxy implements right now is just solomining a single chain.
The idea is that someone can fire up xmrig in daemon mode, and have the proxy inject the necessary merge mining data into the getblocktemplate call.

I will read through your suggestions, thanks.

**Comment by SChernykh** [comment]
Thanks!

All the reference code is in [`src/cryptonote_basic/merge_mining.cpp`](https://github.com/monero-project/monero/blob/master/src/cryptonote_basic/merge_mining.cpp) and https://github.com/monero-project/monero/blob/master/src/rpc/core_rpc_server.cpp#L2070 on the Monero side (plus fixes in https://github.com/monero-project/monero/pull/9073), and in [`src/merge_mining_client.cpp`](https://github.com/SChernykh/p2pool/blob/merge-mining/src/merge_mining_client.cpp), [`src/merkle.cpp`](https://github.com/SChernykh/p2pool/blob/merge-mining/src/merkle.cpp) on P2Pool side.

Documentation is far from ready, but you can ask questions here.

**Comment by parazyd** [comment]
aha I see, so your idea is that the actual full node implements the necessary merge mining methods?

**Comment by SChernykh** [comment]
Yes, the idea is that the full node implements a minimum set of RPCs that will be enough for P2Pool to merge mine.

**Comment by parazyd** [comment]
Will be leaving relevant commits for reference here.

* https://github.com/darkrenaissance/darkfi/commit/82c7ef4e0c81e33fcc75015a26e695f664d4d9ef

**Comment by SChernykh** [comment]
@parazyd Any updates on this?

**Comment by kn0wmad** [comment]
Is this still planned?

**Comment by plowsof** [comment]
🦗
